src/parser.js

```diff
@@ -1001,6 +1001,40 @@ function insertSuperCalls(ast) {
   return ast;
 }
 
+function isArrayAssignmentWithThisReference(path) {
+  const assignedPath = path.value && path.value.left;
+  const isArrayAssignment = assignedPath && assignedPath.type === 'ArrayExpression';
+  const hasMemberExpressions = isArrayAssignment && jsc(assignedPath).find('MemberExpression').size() > 0;
+
+  return isArrayAssignment && hasMemberExpressions;
+}
+
+function insertArrayAssignmentVarDeclarations(assignmentPath) {
+  const identifiers = jsc(assignmentPath.value.left)
+    .find(jsc.Identifier)
+    .filter(path => path.parentPath.name === 'elements')
+    .paths()
+    .map(path => path.value.name)
+    .map(identifierName => jsc.variableDeclarator(jsc.identifier(identifierName), null));
+
+  if (identifiers.length > 0) {
+    jsc(assignmentPath)
+      .closest(jsc.Statement)
+      .insertBefore([
+        jsc.variableDeclaration(
+          'let', identifiers
```

Comment:
Feels like using `var` here is more in line with what role decaf is supposed to play (we can covert these to `let` separately if we need to).